src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.DataConnection.cs

```diff
@@ -1033,10 +1030,10 @@ public List<DataGridViewColumn> GetCollectionOfBoundDataGridViewColumns()
             return cols;
         }
 
-        private void GetSortingInformationFromBackend(out PropertyDescriptor sortProperty, out SortOrder sortOrder)
+        private void GetSortingInformationFromBackend(out PropertyDescriptor? sortProperty, out SortOrder sortOrder)
         {
-            IBindingList ibl = CurrencyManager is not null ? CurrencyManager.List as IBindingList : null;
-            IBindingListView iblv = ibl is not null ? ibl as IBindingListView : null;
+            IBindingList? ibl = CurrencyManager is not null ? CurrencyManager.List as IBindingList : null;
+            IBindingListView? iblv = ibl is not null ? ibl as IBindingListView : null;
 
             if (ibl is null || !ibl.SupportsSorting || !ibl.IsSorted)
```

Comment:
```suggestion
            if (CurrencyManager?.List is not IBindingList { SupportsSorting: true, IsSorted: true } ibl)
```

and use pattern matching at line 1050 for `iblv`